### PR TITLE
Read text node instead of InnerHtml for CSS

### PR DESCRIPTION
**Pull request by patrikwlund**
Because reading `InnerHtml` causes serialization to HTML, with some overhead. Reading the raw data from the text node has no overhead.

A small but easy win for allocations.

#### Before (`master`)
|                 Method |     Mean |   Error |   StdDev |   Median |    Gen0 | Allocated |
|----------------------- |---------:|--------:|---------:|---------:|--------:|----------:|
|     AngleSharpBaseline | 170.2 us | 6.44 us | 19.00 us | 162.8 us |       - | 138.43 KB |
|          MoveCssInline | 826.6 us | 4.63 us | 12.44 us | 825.3 us | 30.0000 | 621.16 KB |
| MoveCssInline_AllFlags | 895.0 us | 5.60 us | 15.90 us | 892.0 us | 30.0000 | 634.99 KB |

#### After this PR
|                 Method |     Mean |   Error |   StdDev |   Median |    Gen0 | Allocated |
|----------------------- |---------:|--------:|---------:|---------:|--------:|----------:|
|     AngleSharpBaseline | 175.7 us | 8.17 us | 24.10 us | 167.3 us |       - | 138.43 KB |
|          MoveCssInline | 833.6 us | 3.50 us |  9.48 us | 832.8 us | 30.0000 | 609.12 KB |
| MoveCssInline_AllFlags | 891.2 us | 4.49 us | 12.15 us | 891.7 us | 30.0000 | 617.41 KB |

### CODE REVIEW

File: PreMailer.Net/PreMailer.Net/Sources/DocumentStyleTagCssSource.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public DocumentStyleTagCssSource(IElement node)
 
 		public string GetCss()
 		{
-			return _node.InnerHtml;
+			return _node.FindChild<IText>()?.Data ?? "";
```
Comment (martinnormark) [inline PreMailer.Net/PreMailer.Net/Sources/DocumentStyleTagCssSource.cs:16]: The way `IText` is found and its Data accessed, is less obvious than `InnerHtml`. To offer the same level of intuitiveness, could it be moved out to an extension method or similar? That would also avoid duplication with code above, as both could use the method.

Such method should be allowed to return `null` and each point of invocation can fall back to their desired value.

**Review by martinnormark — CHANGES_REQUESTED** [review]
Cool! Suggested an extension method. Any thoughts on unit tests that might be relevant here?

File: PreMailer.Net/PreMailer.Net/Sources/DocumentStyleTagCssSource.cs (hunk 1 above)
Comment (patrikwlund) [reply PreMailer.Net/PreMailer.Net/Sources/DocumentStyleTagCssSource.cs]: I added an extension method. The name is very literal and not specific to just CSS, feel free to change it as you like.

Test-wise I think the tests I added for the extension method specifically are enough. The existing tests already verify behavior when style elements are empty for example.

**Comment by patrikwlund** [comment]
As a bonus to the extension method, I also added upped the language version to enable nullable value type support. Nullable is opt-in and can flagged in code with `#nullable enable` and ``#nullable enable``.

I can remove it if you wish to keep it as is or handle it in a separate PR.

**Comment by martinnormark** [comment]
> I also added upped the language version to enable nullable value type support.

How is backward compatibility handled? I assume that any C# codebase below v8.0 would get default nullable behaviour as if the feature was not used at all?

**Comment by patrikwlund** [comment]
> How is backward compatibility handled?

It doesn't change any actual behavior. It's just additional metadata that an analyzer uses to help with warnings/errors, so older consumers won't notice. For example, AngleSharp has nullable support, but this lib didn't need to change anything.

**Review by martinnormark — APPROVED** [review]

**Pull request merged**